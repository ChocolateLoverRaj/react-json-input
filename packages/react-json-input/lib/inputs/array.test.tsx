import arrayInput from './array'
import { render } from '@testing-library/react'

describe('Component', () => {
  test('can add element', () => {
    expect(render(
      <arrayInput.Component
        value={[]}
        name=''
        schema={{ type: 'array' }}
        {...{} as any}
      />
    ).baseElement).toMatchSnapshot()
  })
  test('cannot add element', () => {
    expect(render(
      <arrayInput.Component
        value={[]}
        name=''
        schema={{ type: 'array', maxItems: 0 }}
        {...{} as any}
      />
    ).baseElement).toMatchSnapshot()
  })
})
