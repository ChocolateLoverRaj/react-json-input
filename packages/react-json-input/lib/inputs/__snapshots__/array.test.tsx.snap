// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component can add element 1`] = `
<body>
  <div>
    <tr>
      <td>
        <span>
          ✓
        </span>
      </td>
      <th
        class="react-json-input__lib_InputName.module.scss__inputName"
      />
      <td />
      <td />
      <td />
    </tr>
    <tr>
      <td />
      <th
        class="react-json-input__lib_InputName.module.scss__inputName"
      >
        [+]
      </th>
      <td>
        <button>
          New Element
        </button>
      </td>
    </tr>
  </div>
</body>
`;

exports[`Component cannot add element 1`] = `
<body>
  <div>
    <tr>
      <td>
        <span>
          ✓
        </span>
      </td>
      <th
        class="react-json-input__lib_InputName.module.scss__inputName"
      />
      <td />
      <td />
      <td />
    </tr>
  </div>
</body>
`;
